pkg/madmin/quota-commands.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * MinIO Cloud Storage, (C) 2018 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package madmin
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// QuotaType represents bucket quota type
+type QuotaType string
+
+const (
+	// HardQuota specifies a hard quota of usage for bucket
+	HardQuota QuotaType = "hard"
+	// FIFOQuota specifies a quota limit beyond which older files are deleted from bucket
+	FIFOQuota QuotaType = "fifo"
+)
+
+// IsValid returns true if quota type is one of FIFO or Hard
+func (t QuotaType) IsValid() bool {
+	return t == HardQuota || t == FIFOQuota
+}
+
+// BucketQuota holds bucket quota restrictions
+type BucketQuota struct {
+	Quota uint64    `json:"quota"`
+	Type  QuotaType `json:"quotatype"`
+}
+
+// RemoveBucketQuota - removes quota config on a bucket.
+func (adm *AdminClient) RemoveBucketQuota(ctx context.Context, bucket string) error {
+
+	queryValues := url.Values{}
+	queryValues.Set("bucket", bucket)
+
+	reqData := requestData{
+		relPath:     adminAPIPrefix + "/remove-bucket-quota",
+		queryValues: queryValues,
+	}
+
+	// Execute DELETE on /minio/admin/v3/remove-bucket-quota to delete bucket quota.
+	resp, err := adm.executeMethod(ctx, http.MethodDelete, reqData)
+	defer closeResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusNoContent {
```

Comment:
Thank you @poornas .
Now, `-clear` flag works on the client side with no error messages.
```
(poornas-quota) $ mc admin bucket quota myminio/bucket 64k --hard
_Successfully set bucket quota of 62 KiB with hard type on `bucket`_
(poornas-quota) $ mc admin bucket quota myminio/bucket
_Bucket `bucket` has hard quota of 62 KiB_
(poornas-quota) $ mc admin bucket quota myminio/bucket --clear
_Successfully cleared bucket quota configured on `bucket`_

We know `204` is returned for sure, and that is the fix, but do you guys think we may also keep http.StatusOK in the check?